modules/apps/friendly-url/friendly-url-taglib/src/main/java/com/liferay/friendly/url/taglib/servlet/taglib/InputTag.java

```diff
@@ -182,6 +182,21 @@ private String _getFallbackValue() {
 				return StringPool.BLANK;
 			}
 
+			if (Objects.equals(getClassName(), Layout.class.getName())) {
+				Layout layout = LayoutLocalServiceUtil.fetchLayout(
+					getClassPK());
+
+				if (layout != null) {
+					if (isLocalizable()) {
+						return layout.getFriendlyURLsXML();
+					}
+
+					return layout.getFriendlyURL();
+				}
+
+				return StringPool.BLANK;
+			}
+
```

Comment:
Can you use `_getActualClassName` in the test for consistency?

Also, can you make conditional code a bit flatter? Nested conditionals are harder to understand. Something like:
```
if (Object.equals(_getActualClassname(), ...)) {
    Layout layout = ...;

    if (layout == null) {
        return StringPool.BLANK;
    }

    if (isLocalizable()) {
          ....
    }

    return layout.getFriendlyURLsXML();
```

Reply:
> can you make conditional code a bit flatter? Nested conditionals are harder to understand

Sure thing.

> Can you use _getActualClassName in the test for consistency?

I would not mind, but given that the _getActualClassName makes some additional processing in its implementation precisely with the value to compare, the resulting test that still makes this fix work would be something like:

`if (StringUtil.startsWith(_getActualClassName(), Layout.class.getName())) {`

Would you be ok with this way?

Reply:
Oh, you're right. Then, we should use `getClassName` in both checks; it is pointless to keep using `_getActualClassName` inside `_getFallbackValue()` if it does the wrong thing.